modules/gdscript/gdscript.cpp

```diff
@@ -2639,12 +2639,34 @@ Ref<GDScript> GDScriptLanguage::get_script_by_fully_qualified_name(const String
 
 /*************** RESOURCE ***************/
 
+Ref<Resource> ResourceFormatLoaderGDScript::load_cyclic(const String &p_path, Error *r_error) {
+	Error err = OK;
+	Ref<Resource> res = GDScriptCache::get_shallow_script(p_path, err);
+	if (err != OK) {
```

Comment:
You should always return the actual error code, so this should be `if (r_error)`. Don't presume that the variable was initialized to `OK`.